brayns/common/scene/Model.cpp

```diff
@@ -0,0 +1,284 @@
+                material.second->setSpecularExponent(10.f);
+                material.second->setGlossiness(float(std::rand() % 100) /
+                                               100.f);
+                break;
+            }
+            break;
+        case MaterialsColorMap::shades_of_grey:
+            float value = float(std::rand() % 255) / 255.f;
+            material.second->setDiffuseColor(Vector3f(value, value, value));
+            break;
+        }
+        material.second->commit();
+        ++index;
+    }
+}
+
+void Model::logInformation()
+{
+    uint64_t sizeInBytes = 0;
+    uint64_t nbSpheres = 0;
+    uint64_t nbCylinders = 0;
+    uint64_t nbCones = 0;
+    uint64_t nbMeshes = 0;
+    for (const auto& spheres : _spheres)
+    {
+        sizeInBytes += spheres.second.size() * sizeof(Sphere);
+        nbSpheres += spheres.second.size();
+    }
+    for (const auto& cylinders : _cylinders)
+    {
+        sizeInBytes += cylinders.second.size() * sizeof(Cylinder);
+        nbCylinders += cylinders.second.size();
+    }
+    for (const auto& cones : _cones)
+    {
+        sizeInBytes += cones.second.size() * sizeof(Cones);
+        nbCones += cones.second.size();
+    }
+    for (const auto& trianglesMesh : _trianglesMeshes)
+    {
+        const auto& mesh = trianglesMesh.second;
+        sizeInBytes += mesh.indices.size() * sizeof(Vector3f);
+        sizeInBytes += mesh.normals.size() * sizeof(Vector3f);
+        sizeInBytes += mesh.colors.size() * sizeof(Vector4f);
+        sizeInBytes += mesh.indices.size() * sizeof(Vector3ui);
+        sizeInBytes += mesh.textureCoordinates.size() * sizeof(Vector2f);
+        ++nbMeshes;
+    }
+
+    BRAYNS_INFO << "Spheres: " << nbSpheres << ", Cylinders: " << nbCylinders
+                << ", Cones: " << nbCones << ", Meshes: " << nbMeshes
+                << ", Memory: " << sizeInBytes << " bytes ("
+                << sizeInBytes / 1048576 << " MB), Bounds: " << _bounds
+                << std::endl;
+}
+
+MaterialPtr Model::getMaterial(const size_t materialId)
+{
+    if (_materials.find(materialId) == _materials.end())
+        throw std::runtime_error("Material " + std::to_string(materialId) +
+                                 " is not registered in the model");
+    return _materials[materialId];
+}
+
+void Model::createMissingMaterials()
+{
+    std::set<size_t> materialIds;
+    for (auto& spheres : _spheres)
+        materialIds.insert(spheres.first);
+    for (auto& cylinders : _cylinders)
+        materialIds.insert(cylinders.first);
+    for (auto& cones : _cones)
+        materialIds.insert(cones.first);
+    for (auto& meshes : _trianglesMeshes)
+        materialIds.insert(meshes.first);
+
+    for (const auto materialId : materialIds)
+        try
+        {
+            getMaterial(materialId);
+        }
+        catch (const std::runtime_error&)
+        {
+            createMaterial(materialId, std::to_string(materialId));
+        }
```

Comment:
I would prefer to just do a _materials.find() instead of trying and catching exceptions.

Reply:
The really fishy thing is that getMaterial doesn't return anything. It should have a different name.

Reply:
Fixed. Thanks.